maven/src/org/netbeans/modules/maven/classpath/AbstractProjectClassPathImpl.java

```diff
@@ -206,4 +211,33 @@ public void removePropertyChangeListener(PropertyChangeListener propertyChangeLi
         return project.hashCode() ^ getClass().hashCode();
     }
 
+    /**
+     * Like {@link Artifact#getFile} but when a timestamped snapshot is locally downloaded, uses that instead.
+     */
+    protected static File getFile(Artifact art) {
+        File f = art.getFile();
+        if (f != null) {
+            if (art.isSnapshot() && !art.getVersion().equals(art.getBaseVersion())) {
+                String name = f.getName();
+                int dot = name.lastIndexOf(/* DefaultRepositoryLayout.GROUP_SEPARATOR */'.');
+                if (dot > 0 && name.substring(0, dot).endsWith(art.getBaseVersion())) {
+                    File f2 = new File(f.getParentFile(), name.substring(0, dot - art.getBaseVersion().length()) + art.getVersion() + name.substring(dot));
+                    if (f2.isFile()) {
```

Comment:
An illustrative test for this behavior might be beneficial for future maintenance.

Reply:
Yes…it would take me a while to refresh my memory as to what test facilities exist for creating dummy artifacts. [This comment](https://github.com/apache/incubator-netbeans/blob/4ed13367ea904f6a438d6b3f9624c7555d52226d/maven/test/unit/src/org/netbeans/modules/maven/problems/ProblemReporterImplTest.java#L106) is not encouraging, but it seems there is [some precedent](https://github.com/apache/incubator-netbeans/blob/4ed13367ea904f6a438d6b3f9624c7555d52226d/maven/test/unit/src/org/netbeans/modules/maven/queries/RepositoryForBinaryQueryImplTest.java#L71-L75). In this case, the artifact actually needs to exist in the local repository, _and_ it cannot be something we got from Central, because it is a snapshot.